### ISSUE TITLE
Reader: Improve keyboard navigation accessibility

### ISSUE THREAD

**Issue by tyxla** [Automattic/wp-calypso#61318]
### Quick summary

While reviewing #61112, @pento noticed that there are several accessibility issues with keyboard navigation around the reader. 

This issue quotes @pento's thoughts from https://github.com/Automattic/wp-calypso/pull/61112#pullrequestreview-883931745.

### Steps to reproduce

When tabbing through tab-able elements in the reader list view, those elements can only be triggered by pressing `space`, pressing `enter` opens the currently selected post. However, the equivalent elements can be triggered with `space` or `enter` in the single post view. I noticed this with the Comment and Like buttons, but there may be others.

The currently tabbed element has no relation to the current post in the reader list view. This has two noticeable effects:
- After using `j` to move down a few posts, the tabbed element is now several posts behind, you have to press `tab` a bunch to catch up.
- When you open a single post from the reader list view, then close that post again, the currently tabbed element is reset to the start of the page, but the current post stays as it was prior to it being opened.

### What you expected to happen

Keyboard navigation should be improved.

### What actually happened

This is described in the steps to reproduce

### Context

Discovered by @pento while reviewing #61112.
